.github/actions/set_run_conditions/action.yml

```diff
@@ -0,0 +1,33 @@
+name: 'Set run conditions'
+description: 'Sets the run conditions based on the current GitHub context'
+outputs:
+    conditions:
+        description: 'JSON string of run conditions'
+        value: ${{ steps.set-conditions.outputs.conditions }}
+runs:
+    using: 'composite'
+    steps:
+        - id: set-conditions
+          run: |
+              # Define conditions using shell variables
+              IS_MAIN_PUSH=${{ github.ref == 'refs/heads/main' && !contains(github.event.head_commit.message, 'demergent-labs/release--') }}
+              IS_MAIN_MERGE_RELEASE=${{ github.ref == 'refs/heads/main' && contains(github.event.head_commit.message, 'demergent-labs/release--') }}
+              IS_RELEASE_PR=${{ startsWith(github.head_ref, 'release--') }}
+              IS_FEATURE_PR=${{ !startsWith(github.head_ref, 'release--') && github.ref != 'refs/heads/main' && github.event.pull_request.draft == false }}
```

Comment:
Do you need to check `refs/heads/main`?

Reply:
maybe not, it's what was here when I got here. I mean we don't want this to be true if the branch is main right?